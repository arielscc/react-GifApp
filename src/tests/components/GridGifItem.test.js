const { shallow } = require("enzyme");


const { default: GridGifItem } = require("../../components/GridGifItem");

import React from 'react'
import '@testing-library/jest-dom'

const datos = {
  title : 'Titulo del test',
  url : 'https://algunaurl.com'
}
const wrapper = shallow( <GridGifItem { ...datos} /> )

describe('Probando el componente Grid Gif Item', () => {
  test('El componente debe hacer match con el snapshot', () => {
    expect( wrapper ).toMatchSnapshot()
  });

  test('Debe de tener un titulo', () => {
    const title = wrapper.find( 'h3' );
    expect( title.text().trim() ).toBe( datos.title );
  });

  test('Debe de tener un src y alt', () => {
    const img = wrapper.find( 'img' );
    expect(img.prop('src') ).toBe( datos.url );
    expect(img.prop('alt') ).toBe( datos.title );
  });

  test('debe de tener la clase animate__slideInUp', () => {
    const className = 'animate__slideInUp';
    const div = wrapper.find( 'div' )
    expect( div.hasClass( className ) ).toBe( true )
  });
});