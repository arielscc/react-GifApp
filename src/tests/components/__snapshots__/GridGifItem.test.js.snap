// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente Grid Gif Item El componente debe hacer match con el snapshot 1`] = `
<div
  className="card animate__animated animate__slideInUp"
>
  <img
    alt="Titulo del test"
    className="images"
    src="https://algunaurl.com"
  />
  <h3>
     
    Titulo del test
     
  </h3>
</div>
`;
